plugin/trino-druid/src/main/java/io/trino/plugin/druid/aggregate/SingleInputAggregateFunction.java

```diff
@@ -0,0 +1,134 @@
+import static io.trino.plugin.jdbc.expression.AggregateFunctionPatterns.singleInput;
+import static io.trino.plugin.jdbc.expression.AggregateFunctionPatterns.variable;
+import static java.util.Objects.requireNonNull;
+
+//TODO Move this class to base-jdbc module and update all jdbc connectors to leverage it.
+public class SingleInputAggregateFunction
+        implements AggregateFunctionRule
+{
+    private static final Capture<Variable> INPUT = newCapture();
+    private final String prestoName; // presto aggregate function name to match
+    private final Optional<String> expression; // the expression format
+    private final Optional<JdbcTypeHandle> jdbcTypeHandle; // type handle if its different from input column
+    private Type inputType; // empty if all input types should match, or the set to match
+    private Type outputType; // provide if the pattern should only match specific output type
+
+    public SingleInputAggregateFunction(
+            String prestoName,
+            Optional<String> expression,
+            Optional<JdbcTypeHandle> jdbcTypeHandle,
+            Type inputType,
+            Type outputType)
+    {
+        this.prestoName = requireNonNull(prestoName, "prestoName is null");
+        this.expression = requireNonNull(expression, "expression is null");
+        this.jdbcTypeHandle = requireNonNull(jdbcTypeHandle, "jdbcTypeHandle is null");
+        this.inputType = requireNonNull(inputType, "inputType is null");
+        this.outputType = requireNonNull(outputType, "outputType is null");
+    }
+
+    @Override
+    public Pattern<AggregateFunction> getPattern()
+    {
+        return basicAggregation()
+                .with(functionName().equalTo(prestoName))
+                .with(outputType().equalTo(outputType))
+                .with(singleInput()
+                        .matching(variable().with(expressionType().equalTo(inputType)).capturedAs(INPUT)));
+    }
+
+    @Override
+    public Optional<JdbcExpression> rewrite(AggregateFunction aggregateFunction, Captures captures, RewriteContext context)
+    {
+        Variable input = captures.get(INPUT);
+        JdbcColumnHandle columnHandle = (JdbcColumnHandle) context.getAssignments().get(input.getName());
+        verifyNotNull(columnHandle, "Unbound variable: %s", input);
+        String expressionFormat = expression.orElse(prestoName + "(%s)");
+        return Optional.of(new JdbcExpression(
+                expressionFormat.replaceAll("%s", columnHandle.getColumnName()),
+                jdbcTypeHandle.orElse(columnHandle.getJdbcTypeHandle())));
+    }
+
+    public static Builder builder()
+    {
+        return new Builder();
+    }
+
+    public static final class Builder
+    {
+        private String prestoName;
+        private Optional<String> expression = Optional.empty();
+        private Optional<JdbcTypeHandle> jdbcTypeHandle = Optional.empty();
+        private Type inputType;
+        private Type outputType;
```

Comment:
sort fields and setters

1. what defines match (name, input type, output type)
2. how it's transformed

Reply:
Done